src/main/java/org/sonarlint/intellij/SonarLintIntelliJClient.kt

```diff
@@ -137,6 +140,47 @@ object SonarLintIntelliJClient : SonarLintRpcClientDelegate {
         suggestionsByConfigScopeId.forEach { (configScopeId, suggestions) -> suggestAutoBind(findProject(configScopeId), suggestions) }
     }
 
+    override fun suggestConnection(suggestionsByConfigScope: Map<String, List<ConnectionSuggestionDto>>) {
+        for (suggestion in suggestionsByConfigScope) {
+            val project = BackendService.findModule(suggestion.key)?.project
+                ?: BackendService.findProject(suggestion.key) ?: return
+
+            if (suggestion.value.size == 1) {
+                //It was decided to only handle the case where there is only one notification per configuration scope
+                val uniqueSuggestion = suggestion.value[0]
+
+                val (connectionKind, projectKey, connectionName) = getAutoShareConfigParams(uniqueSuggestion)
+
+                ConfigurationSharing.showAutoSharedConfigurationNotification(
+                    project, String.format(
+                        """
+                    A Connected Mode configuration file is available to bind to project "%s" on %s "%s".
+                    You can also configure the binding manually later.
+                """.trimIndent(), projectKey, connectionKind, connectionName
+                    ), SKIP_AUTO_SHARE_CONFIGURATION_DIALOG_PROPERTY
+                )
+            }
+        }
+    }
+
+    private fun getAutoShareConfigParams(uniqueSuggestion: ConnectionSuggestionDto): Triple<String, String, String> {
+        val connectionKind: String
+        val projectKey: String
+        val connectionName: String
+
+        if (uniqueSuggestion.connectionSuggestion.isRight) {
+            connectionKind = "SonarCloud organization"
+            projectKey = uniqueSuggestion.connectionSuggestion.right.projectKey
+            connectionName = uniqueSuggestion.connectionSuggestion.right.organization
```

Comment:
You can directly return a `Triple()` inside the if and else, so you don't have to declare the variables